lms/resources/_js_config/__init__.py

```diff
@@ -87,6 +87,14 @@ def add_document_url(self, document_url):
                     _query={"document_url": document_url},
                 ),
             }
+
+        elif document_url.startswith("canvas://"):
+            self.add_canvas_file_id(
+                self._request.params["custom_canvas_course_id"],
+                self._request.params["resource_link_id"],
+                self._request.params["file_id"],
```

Comment:
Note that after all of this we are still using the param from the URL and not from document_url. 

I think we should do that on a separate URL and refactor `canvas_api.files.via_url` view to take the full URL instead of the separate parts, as the blackboard one does.